File: backend/src/main/scala/microservices/blog/objects/CreateBlogPostPlannerRequest.scala
// CreateBlogPostPlannerRequest：博客域创建博客文章请求对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateBlogPostPlannerRequest(
    userId: String,
    title: String,
    summary: String,
    content: String,
    images: List[BlogImagePlannerResponse],
    tags: Option[List[BlogTagPlannerResponse]],
    coverText: Option[String],
    travelCity: Option[String],
    travelCities: Option[List[String]]
)
object CreateBlogPostPlannerRequest:
  given sourceEncoder: Encoder[CreateBlogPostPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateBlogPostPlannerRequest] = deriveDecoder

File: frontend/src/microservices/blog/objects/CreateBlogPostPlannerRequest.ts
// CreateBlogPostPlannerRequest：博客域创建博客文章请求对象。

import type { ContentImagePlannerResponse } from '@/microservices/content/objects/ContentImagePlannerResponse'
import type { BlogTagResponse } from './BlogTagResponse'

export type CreateBlogPostPlannerRequest = {
  userId: string
  title: string
  summary: string
  content: string
  images: ContentImagePlannerResponse[]
  tags?: BlogTagResponse[]
  coverText?: string
  travelCity?: string | null
  travelCities?: string[] | null
}

export const createBlogPostPlannerRequestFromJson = (json: string): CreateBlogPostPlannerRequest =>
  JSON.parse(json) as CreateBlogPostPlannerRequest

export const createBlogPostPlannerRequestToJson = (value: CreateBlogPostPlannerRequest): string =>
  JSON.stringify(value)